conda-store-server/tests/test_server.py

```diff
@@ -335,6 +336,58 @@ def test_create_specification_unauth(testclient):
     assert r.status == schema.APIStatus.ERROR
 
 
+# Only testing size values that will always cause errors. Smaller values could
+# cause errors as well, but would be flaky since the test conda-store state
+# directory might have different lengths on different systems, for instance,
+# due to different username lengths.
+@pytest.mark.parametrize(
+    "size",
+    [
+        # OSError: [Errno 36] File name too long
+        255,
+        # OSError: [Errno 36] File name too long
+        256,
+    ],
+)
+def test_create_specification_auth_env_name_too_long(testclient, celery_worker, authenticate, size):
+    namespace = "default"
+    environment_name = 'A' * size
+
+    response = testclient.post(
+        "api/v1/specification",
+        json={
+            "namespace": namespace,
+            "specification": json.dumps({"name": environment_name}),
+        },
+    )
+    response.raise_for_status()
+
+    r = schema.APIPostSpecification.parse_obj(response.json())
+    assert r.status == schema.APIStatus.OK
+
+    # Try checking that the status is 'FAILED'
+    is_updated = False
+    for _ in range(5):
+        time.sleep(5)
+
+        # check for the given build
+        response = testclient.get(f"api/v1/build/{r.data.build_id}")
```

Comment:
This is a bug: it should copy `build_id` from the `r` above since `r` is reused in a loop. See the other test that does this right.